amr-wind/ocean_waves/relaxation_zones/stokes_waves_K.H

```diff
@@ -4,8 +4,44 @@
 #include <AMReX_FArrayBox.H>
 #include <cmath>
 
+// Stokes waves theory adapted from
+// Fenton, J., Fifth Order Stokes Theory for Steady Waves
+// Journal of Waterway, Port, Coastal and Ocean Engineering, 1985, 111, 216-234
+
 namespace amr_wind::ocean_waves::relaxation_zones {
 
+// Compute wavelength as a function of wave period, water depth, and g
+AMREX_FORCE_INLINE amrex::Real stokes_wave_length(
+    const amrex::Real T, const amrex::Real d, const amrex::Real g)
+{
+    const amrex::Real omega = 2.0 * M_PI / T;
+    const amrex::Real lambda0 = (g * T * T) / 2.0 / M_PI;
+
+    // Begin Newton-Raphson iterations
+    constexpr amrex::Real tol = 1e-10;
+    constexpr int iter_max = 20;
+    int iter = 0;
+    amrex::Real k = omega / std::sqrt(g);
+    amrex::Real f = g * k * std::tanh(k * d) - omega * omega;
+    while (f > 1e-10 && iter < iter_max) {
```

Comment:
was this 1e-10 meant to use `tol` above? In general should we have a more uniform way of specifying tolerances. Feels like we have this and `eps` in a bunch of different places. Not necessarily a "now" thing...

Reply:
thanks for noticing that. I agree that it would be good to have a more standardized reference for tolerances, but often the level of tolerance required depends on the context. So I would think that ad-hoc tolerances would still be used.